pkg/util/machine/azureproviderspec.go

```diff
@@ -0,0 +1,52 @@
+package machine
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	machinev1beta1 "github.com/openshift/api/machine/v1beta1"
+	"k8s.io/client-go/kubernetes/scheme"
+
+	_ "github.com/Azure/ARO-RP/pkg/util/scheme"
+)
+
+type MachineType string
+
+const (
+	MachineSet MachineType = "machineset"
+	Machine    MachineType = "machine"
+)
+
+// UnmarshalAzureProviderSpec unmarshals an Azure Provider Spec set on machines or machinesets
+// it contains backward compatibility for the older azureproviderconfig which is no longer in use
+// in OCP 4.10+
+func UnmarshalAzureProviderSpec(name string, mType MachineType, rawProviderSpec []byte) (*machinev1beta1.AzureMachineProviderSpec, error) {
+	var machineProviderSpec *machinev1beta1.AzureMachineProviderSpec
+
+	if strings.Contains(string(rawProviderSpec), "azureproviderconfig.openshift.io") {
```

Comment:
I would extract 'azureproviderconfig.openshift.io' into a constant as it is used in two places (here and in line 34).